hal/source/bmp180_device.cpp

```diff
@@ -0,0 +1,326 @@
+#include "bmp180_device.h"
+#include <thread>
+
+Bmp180Device::Bmp180Device():
+	serial_(protocol_),
+	write_addr_(default_i2c_addr_),
+	read_addr_(default_i2c_addr_ | 0x01) {
+}
+
+Bmp180Device::~Bmp180Device() {
+
+}
+
+error_condition Bmp180Device::ReConfig(const CommonKeyPairs& config) {
+	ClearInitialized();
+	auto key_value = config.find(kPortName);
+	if (key_value == config.end())
+		return kInvalidArgument;
+	else {
```

Comment:
No need to have this `else`. The "early return" coding style is for eliminating indentation.